File: core-java-practice/leet-code-codebase/palindrome.java
//package core-java-practice.leet-code-codebase;
import java.util.*;
public class palindrome {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int a=s.nextInt();
        int reversedNumber=0;
        if(a<0||(a%10==0 && a!=0)){     //base case check karenge
            System.out.println("Not a palindrome");
            return;
        }
        while(a>reversedNumber){
            reversedNumber=reversedNumber*10+a%10; /*ek half of no. reverseNumber mai daal denge*/
            a=a/10;
        }
        if(a==reversedNumber||a==reversedNumber/10){   /*dono half compare karenge*/
            System.out.println("Number is palindrome");
        }
        s.close();

    }
}

File: core-csharp-practice/leet-code-practice/Palindrome.cs
using System;

class Palindrome
{
    static void Main()
    {
        int a = 121;   // fixed number to check

        int reversedNumber = 0;

        // base case check
        if (a < 0 || (a % 10 == 0 && a != 0))
        {
            Console.WriteLine("Not a palindrome");
            return;
        }

        // reverse only half of the number
        while (a > reversedNumber)
        {
            reversedNumber = reversedNumber * 10 + a % 10;
            a = a / 10;
        }

        // compare both halves
        if (a == reversedNumber || a == reversedNumber / 10)
        {
            Console.WriteLine("Number is palindrome");
        }
        else
        {
            Console.WriteLine("Not a palindrome");
        }
    }
}
